### DISCUSSION TITLE
OR tools VRP: how to make a constraint to ensure a vehicle loading goods satifying both weight and cubic meters limits?

### DISCUSSION THREAD

**Discussion by Westlife1002** [google/or-tools#2615] [Routing (and legacy CP) questions] [answered]
I have developed a VRP model based on ortools/vrp package. the full code is here:
https://gist.github.com/Westlife1002/d920226616951d2b14d91e25eb9f23a1 

Previously I have only considered "weight" as a single dimention for load consolidation and route planning. To be realistic, the VRP planning should consider both "weight" and "cubic meters" for consolidating loads for a vehicle. Hence, how do I make a constraint to ensure a vehicle loading goods satifying both conditions? Are there any examples works in real practice, particularly considering the complex situation of the various size/shape of the goods putting together in the same truck.

**Comment by Mizux** [comment, marked as answer]
Trying to solve a 3D bin packing problem each time you add a new location to the vehicle's route won't be possible and you'll just kill the performance of the routing search near to zero.

BUT you could add a second dimension like the load dimension but this time with volume aka m^3 instead of mass aka kg.
Then once a solution is found, you can try to check this solution if all goods will actually fit.
tips: instead of taking the maximum available volume of the vehicle, use ~90% of it etc so you have room if it not perfectly fit...
